crates/rpc/rpc/src/eth/sim_bundle.rs

```diff
@@ -335,7 +336,21 @@ where
                     // TODO: since we are looping over iteratively, we are not collecting bundle
                     // logs. We should collect bundle logs when we are processing the bundle items.
                     if logs {
-                        let tx_logs = result.logs().to_vec();
+                        let tx_logs = result
+                            .logs()
+                            .iter()
+                            .enumerate()
+                            .map(|(log_index, log)| alloy_rpc_types_eth::Log {
+                                inner: log.clone(),
+                                block_hash: None,
+                                block_number: None,
+                                block_timestamp: None,
+                                transaction_hash: Some(*item.tx.tx_hash()),
+                                transaction_index: Some(tx_index as u64),
+                                log_index: Some(log_index as u64),
```

Comment:
isn't this the index in the block? so we should keep a counter across the tx_index loop?

Reply:
Yup good call, fixed in https://github.com/paradigmxyz/reth/pull/14506/commits/be44fa823b592bd36d16bc48a13b77fedb57c651